### PR TITLE
Better error message, when backup path is a directory and not a file

### LINKED ISSUE #869: Support backups to directory
- what were your actions?
I want to create al local backup of my Database
- what was wrong?
When i click on "save now" i got ERROR: EISDIR illegal operation on a direcoty, open a "Path to my backup folder
- app version
Desktopapp V 1.6.3
- your user-agent (from Settings/Help section)
Mozilla/5.0 (Windows NT 10.0; Win64; x64) 
- does it happen on Demo or New database?
old Database from Keepass2
- please, open dev tools in your browser and attach output log from Console tab 
    (if you are using desktop app, devtools can be opened from Settings/General/Advanced)
2018-01-21T17:19:21.803Z [backup:Florian] Backup file to file "Path to my backup folder"
index.html:1 2018-01-21T17:19:21.804Z [storage-file] Stat "Path from my Databse" 
index.html:1 2018-01-21T17:19:21.805Z [storage-file] Stat done "Path from my Databse"   1ms
index.html:1 2018-01-21T17:19:21.805Z [backup:Florian] Backup folder exists, saving
index.html:1 2018-01-21T17:19:21.806Z [storage-file] Save "Path to my backup folder" undefined
index.html:1 2018-01-21T17:19:21.807Z [storage-file] Error writing local file "Path to my backup folder" Error: EISDIR: illegal operation on a directory, open 'Path to my backup folder'
a.error @ index.html:1
l @ index.html:1
(anonymous) @ index.html:1
(anonymous) @ fs.js:1306
(anonymous) @ fs.js:114
index.html:1 2018-01-21T17:19:21.811Z [backup:Florian] Backup error Error: EISDIR: illegal operation on a directory, open 'Path to my backup folder'

### PR DESCRIPTION
**Pull request by dploeger**
closes #869

### CODE REVIEW

File: app/scripts/locales/fr-FR.json [hunk 1]
```diff
@@ -426,6 +426,8 @@
   "setFileBackupNowWorking": "En cours...",
   "setFileBackupError": "Erreur sauvegarde",
   "setFileBackupErrorDescription": "Erreur à l'écriture du fichier",
+  "setFileBackupErrorIsDir": "Chemin de sauvegarde invalide",
```
Comment (antelle) [inline app/scripts/locales/fr-FR.json]: Please don't edit these files: they come from OneSky and will be overwritten in any case.

File: app/scripts/views/settings/settings-file-view.js [hunk 2]
```diff
@@ -512,10 +512,22 @@ const SettingsFileView = Backbone.View.extend({
                 this.backupInProgress = false;
                 backupButton.text(Locale.setFileBackupNow);
                 if (err) {
+                    let title = ''
```
Comment (antelle) [inline app/scripts/views/settings/settings-file-view.js]: Please add commas. You can also validate violations with eslint to see all of them.

File: app/scripts/views/settings/settings-file-view.js [hunk 3]
```diff
@@ -512,10 +512,22 @@ const SettingsFileView = Backbone.View.extend({
...
+                    let description = ''
+                    if (err.hasOwnProperty('code') && err.code === 'EISDIR') {
```
Comment (antelle) [inline app/scripts/views/settings/settings-file-view.js]: The error code is node-specific: it shouldn't be here. Please move to storage implementation and return as a property of error from there.

File: app/scripts/locales/fr-FR.json (hunk 1 above)
Comment (dploeger) [reply app/scripts/locales/fr-FR.json]: Oh, okay. Didn't know that. So just add something to the base.json and the rest will come from OneSky?

File: app/scripts/views/settings/settings-file-view.js (hunk 3 above)
Comment (dploeger) [reply app/scripts/views/settings/settings-file-view.js]: You mean like modify the returned error object in i.e. StorageFile#save to include .isDir=true when code === EISDIR and something like that for the other possible storage providers?

File: app/scripts/views/settings/settings-file-view.js (hunk 2 above)
Comment (dploeger) [reply app/scripts/views/settings/settings-file-view.js]: Whoops, sorry. Forgot to enable eslint in my IDE. Fixed.

File: app/scripts/locales/fr-FR.json (hunk 1 above)
Comment (antelle) [reply app/scripts/locales/fr-FR.json]: Yes, correct.

File: app/scripts/views/settings/settings-file-view.js (hunk 3 above)
Comment (antelle) [reply app/scripts/views/settings/settings-file-view.js]: Yes exactly. To move node fs dependencies into fs provider.

Comment (dploeger) [reply app/scripts/views/settings/settings-file-view.js]: ✅

**Comment by antelle** [comment]
Thank you! 🎁

**Pull request merged**